code/modules/overmap/ships/computers/sensors.dm

```diff
@@ -176,58 +217,53 @@
 	else
 		data["id_status"] = "NOBEACON" //Should not really happen.
 
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
-	if (!ui)
-		ui = new(user, src, ui_key, "shipsensors.tmpl", "[linked.get_real_name()] Sensors Control", 600, 530, src)
-		ui.set_initial_data(data)
-		ui.open()
-		ui.set_auto_update(1)
+	return data
 
-/obj/machinery/computer/ship/sensors/Topic(href, href_list)
+/obj/machinery/computer/ship/sensors/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
 	if (..())
 		return TOPIC_HANDLED
 
 	if (!linked)
 		return TOPIC_NOACTION
 
-	if (href_list["viewing"])
+	if (action == "viewing")
 		if(usr)
 			viewing_overmap(usr) ? unlook(usr) : look(usr)
 		return TOPIC_REFRESH
 
-	if (href_list["link"])
+	if (action == "link")
 		find_sensors_and_iff()
 		return TOPIC_REFRESH
 
 	if(sensors)
-		if (href_list["range"])
+		if (action == "range")
 			var/nrange = input("Set new sensors range", "Sensor range", sensors.range) as num|null
 			if(!CanInteract(usr, default_state))
 				return TOPIC_NOACTION
 			if (nrange)
 				sensors.set_desired_range(Clamp(nrange, 1, sensors.max_range))
 			return TOPIC_REFRESH
```

Comment:
These returns are not following TGUI's logic, TGUI's logic is as follows:
```
/**
 * public
 *
 * Called on a UI when the UI receieves a href.
 * Think of this as Topic().
 *
 * required action string The action/button that has been invoked by the user.
 * required params list A list of parameters attached to the button.
 *
 * return bool If the user's input has been handled and the UI should update.
 */
```
The defines used here are for nanoui, and should be ported to use TGUI's logic, as follows:

TOPIC_HANDLED / TOPIC_REFRESH -> TRUE
TOPIC_NOACTION -> FALSE

Returning TRUE from `ui_act` causes the TGUI interface to update, FALSE to not update

Reply:
fixed in most recent commit